source/vecs/entity.d

```diff
@@ -362,15 +427,17 @@ public:
 
 
 	/**
-	 * Removes all components from an entity. It searches every Component
-	 *     registered and checks wether or not the entity is asociated with it,
-	 *     if it is, it removes the component, otherwise it ignores it and
-	 *     continues to the next. If an invalid entity is passed it leads to
-	 *     undefined behaviour.
+	 * Removes all components associated to an entity. If the entity passed is
+	 *     invalid it leads to undefined behaviour.
+	 *
+	 * Safety: The **internal code** is @safe, however, beacause of **Signal**
+	 *     dependency, the method must be @system.
+	 *
+	 * Signal: Emits onRemove.
 	 *
 	 * Params: e = an entity to disassociate.
 	 */
-	@safe pure
+	@system
```

Comment:
Isn't this isolated from the signal? This can be `@safe` as it doesn't depend on the user callbacks. Other functions can be `@safe` too, if it doesn't directly depend on the user code.

Reply:
It does depend on Signal. As the documentation states, this will emit onRemove.